embassy-nrf/src/util.rs

```diff
@@ -2,14 +2,14 @@ const SRAM_LOWER: usize = 0x2000_0000;
 const SRAM_UPPER: usize = 0x3000_0000;
 
 /// Does this slice reside entirely within RAM?
-pub(crate) fn slice_in_ram(slice: &[u8]) -> bool {
+pub(crate) fn slice_in_ram<T>(slice: &[T]) -> bool {
```

Comment:
if you allow arbitrary types, you need `slice.len() * size_of::<T>()`

Reply:
Like this? 
```
ptr >= SRAM_LOWER && (ptr + slice.len() * core::mem::size_of::<T>()) < SRAM_UPPER
```

Reply:
:+1: